src/main/java/org/broadinstitute/hellbender/utils/recalibration/BaseRecalibrationEngine.java

```diff
@@ -120,11 +124,18 @@ public void processRead( final GATKRead originalRead, final ReferenceDataSource
         int[] isSNP = new int[read.getLength()];
         int[] isInsertion = new int[isSNP.length];
         int[] isDeletion = new int[isSNP.length];
-        final int nErrors = calculateIsSNPOrIndel(read, refDS, isSNP, isInsertion, isDeletion);
 
-        // note for efficiency regions we don't compute the BAQ array unless we actually have
-        // some error to marginalize over.  For ILMN data ~85% of reads have no error
-        final byte[] baqArray = nErrors == 0 ? flatBAQArray(read) : calculateBAQArray(read, refDS);
+        final byte[] baqArray;
+
+        //Note: this function modifies the isSNP, isInsertion and isDeletion arguments so it can't be skipped, BAQ or no BAQ
+        final int nErrors = calculateIsSNPOrIndel(read, refDS, isSNP, isInsertion, isDeletion);
+        if (recalArgs.skipBAQ){
+            baqArray = flatBAQArray(read);
+        } else {
+            // note for efficiency reasons we don't compute the BAQ array unless we actually have
+            // some error to marginalize over.  For ILMN data ~85% of reads have no error
+            baqArray = nErrors == 0 ? flatBAQArray(read) : calculateBAQArray(read, refDS);
+        }
```

Comment:
Simplify the above to the one-liner:

    final byte[] baqArray = (nErrors == 0 || recalArgs.skipBAQ) ? flatBAQArray(read) : calculateBAQArray(read, refDS);